### PR TITLE
feat: fix quality criteria issues gf-382

### PR DESCRIPTION
**Pull request by Vitaliistf**
Fixed quality criteria issues in mentioned files. Also looked through the whole project to find other issues like these.

### CODE REVIEW

**Comment by Vitaliistf** [comment]
While looking through the code, found some things that might be improved, in my opinion. I'm not sure about it so please, let me know if I should change this code.

1. I think we can simplify this, remove `hasError` constant and use `error` to check if it's truthy.
Screenshot 2024-09-18 at 15 39 16

2. Menu item component is intended to be used in menu, so we could move it to menu's components and re-export it there.
Screenshot 2024-09-18 at 15 43 58

**Comment by GvoFor** [comment]
1. We are trying to stick to the following approach: in the constructions like `{condition && <component>}` condition type should be boolean, so that's why `hasError` used here. I think you shouldn't remove it.

2. I'm not sure, but I'd like to suggest the following: 
- `MenuItem` should be local component relative to `Menu`
- Others components shouldn't know about `MenuItem` -- there should be some type describing `MenuItem` props, and we should pass an array of such a type to `Menu` component, and `Menu` is responsible to convert this array into array of `MenuItem` objects (which it will render). So I image menu usage as:
```ts
<Menu
  isOpened=...
  onClose=...
  onOpen=...
  items={[
    { icon: ..., label: ..., ... }
  ]}
/>
```

**Comment by liza-veis** [comment]
@GvoFor I agree with your approach, but for now let's leave it as it is, at least in this ticket

File: apps/frontend/src/libs/components/table/styles.module.css [hunk 1]
```diff
@@ -65,7 +65,7 @@
 	table-layout: fixed;
 }
 
-.table .table-row:last-child .table-data {
+.table-row:last-child .table-data {
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/styles.module.css:68]: Maybe we need to remove .table for all other selectors here to keep it consistent?

Comment (Vitaliistf) [reply apps/frontend/src/libs/components/table/styles.module.css:68]: Yeah, makes sense. Thank you!

**Comment by perekotypole** [comment]
Could you move `apps\backend\src\modules\project-groups\libs\helpers\generate-project-group-key.helper.ts` as well as you did with `getValidClassNames` if we keep helpers in folders everywhere?

**Comment by Vitaliistf** [comment]
> Could you move `apps\backend\src\modules\project-groups\libs\helpers\generate-project-group-key.helper.ts` as well as you did with `getValidClassNames` if we keep helpers in folders everywhere?

Sure, good notice, Thanks!

**Review by what1s1ove — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**